build.sc

```diff
@@ -8,7 +8,7 @@ import com.github.lolgab.mill.mima._
 
 val scala212 = "2.12.17"
 val scala213 = "2.13.10"
-val scala3 = "3.1.3"
+val scala3 = "3.3.1"
```

Comment:
Is this intentional? We drop support for older Scala 3 version, which might be Ok, but should be done in a new `0.x` bump.

Reply:
It's a compiler plugin, so will need separate implementations for each Scala version I think. 

It's possible to support 3.1.3, but the initial proof of concept is only against 3.3.1. Will have to decide whether or not to drop 3.1.3 before merging this, but I'd be in favor just to support the 3.3.x LTS versions